src/main/java/eu/europeana/iiif/service/EdmManifestMapping.java

```diff
@@ -610,17 +612,24 @@ private static String getLicenseText(String europeanaId, Object jsonDoc) {
     /**
      * Generates a new canvas, but note that we do not fill the otherContent (Full-Text) here. That is done later
      */
-    static eu.europeana.iiif.model.v2.Canvas getCanvasV2(ManifestSettings settings,
-                                                                 String europeanaId,
+    private static eu.europeana.iiif.model.v2.Canvas getCanvasV2(String europeanaId,
                                                                  int order,
                                                                  WebResource webResource,
                                                                  Map<String, Object>[] services) {
-        eu.europeana.iiif.model.v2.Canvas c = new eu.europeana.iiif.model.v2.Canvas(Definitions.getCanvasId(europeanaId, order),
-                order, settings.getCanvasHeight(), settings.getCanvasWidth());
+        eu.europeana.iiif.model.v2.Canvas c =
+                new eu.europeana.iiif.model.v2.Canvas(Definitions.getCanvasId(europeanaId, order), order);
 
         c.setLabel("p. "+order);
 
-        String attributionText = (String) webResource.get("textAttributionSnippet");
+        if (safeToCast(webResource, EBUCOREHEIGHT)){
```

Comment:
This will work, but I do feel the code is a bit too complicated unnecessarily. Why not use
```
        Object o = webResource.get(EBUCOREHEIGHT);
        if (o instanceof Integer){
            c.setHeight((Integer) o);
        }
```
		
This is shorter and saved you from doing a null-check and retrieving the property 3 times from the webresource hashmap (although the later is probably be optimized by the compiler)